### DISCUSSION TITLE
Please add Edit icon in Action

### DISCUSSION THREAD

**Discussion by paolostefanucci** [prasathmani/tinyfilemanager#877] [Ideas]
I use tinyfilemanager to code in PHP with browser from anywhere.
I direct edit php files on the server.
It would be great to have an Edit icon (better an Advanced Editor Icon) in the Actions icons for .html, .htm, .js, .css, .php, .txt, .json files
It is a waste of time to click on the .php file, wait to view the file in a new page and than click on "Advanced Editor".
Maybe the best solution is to use the Preview Icon (the first actions icon) only for images, video, etc. and substitute the "eye" icon with an Advanced Editor Icon for editable files likes .html, .htm, .js, .css, .php, .txt, .json 

Thank you in advance.
P.S.
Tinyfilemanager is a great piece of software, thank you for made it open source.

**Comment by prasathmani** [comment]
@paolostefanucci ,Thank you for your valuable input, I will add to the feature releases.

**Comment by paolostefanucci** [comment]
Thank you so much

**Comment by aleddrj** [comment]
Ciao @paolostefanucci, while @prasathmani i finds the time to make the modification properly, I'll give you a snippet of the code ;-)
```
2267	<td class="inline-actions">
2268		<?php if (!FM_READONLY): ?>
2269			<a title="<?php echo lng('Delete') ?>" href="?p=<?php echo urlencode(FM_PATH) ?>&amp;del=<?php echo urlencode($f) ?>" onclick="confirmDailog(event, 1209, '<?php echo lng('Delete') . ' ' . lng('File'); ?>','<?php echo urlencode($f); ?>', this.href);"> <i class="fa fa-trash-o"></i></a>
2270			<a title="<?php echo lng('Rename') ?>" href="#" onclick="rename('<?php echo fm_enc(addslashes(FM_PATH)) ?>', '<?php echo fm_enc(addslashes($f)) ?>');return false;"><i class="fa fa-pencil-square-o"></i></a>
2271			<a title="<?php echo lng('CopyTo') ?>..."
2272			href="?p=<?php echo urlencode(FM_PATH) ?>&amp;copy=<?php echo urlencode(trim(FM_PATH . '/' . $f, '/')) ?>"><i class="fa fa-files-o"></i></a>
2273		<?php endif; ?>
```
questa è la riga di codice da inserire dopo la riga 2273
`	<a title="<?php echo lng('AdvancedEditor') ?>" href="?p=<?php echo urlencode(trim(FM_PATH)) ?>&amp;edit=<?php echo urlencode(trim($f, '/')) ?>&env=ace" class="edit-file"><i class="fa fa-pencil-square"></i></a>`

**Reply by paolostefanucci** [reply]
Thank you so much.
